steppy/base.py

```diff
@@ -376,8 +405,56 @@ def transform(self, data):
             else:
                 step_inputs = self._unpack(step_inputs)
             step_output_data = self._transform_operation(step_inputs)
+        logger.info('Step {}, transform completed'.format(self.name))
         return step_output_data
 
+    def set_mode_train(self):
+        """Applies 'train' mode to all upstream Steps including this Step
+        and cleans cache for all upstream Steps including this Step.
+        """
+        self._set_mode('train')
+
+    def set_mode_inference(self):
+        """Applies 'inference' mode to all upstream Steps including this Step
+        and cleans cache for all upstream Steps including this Step.
+        """
+        self._set_mode('inference')
+
+    def reset(self):
+        """Reset all upstream Steps to the default training parameters and
+        cleans cache for all upstream Steps including this Step.
+        Defaults are:
+            'mode': 'train',
+            'is_fittable': True,
+            'force_fitting': True,
+            'persist_output': False,
+            'cache_output': False,
+            'load_persisted_output': False,
+            'persist_upstream_pipeline_structure': False
+        """
+        self.clean_cache_upstream_steps()
+        self.set_mode_train()
+        for name, step_obj in self.all_upstream_steps.items():
+            step_obj.is_fittable = DEFAULT_TRAINING_SETUP['is_fittable']
+            step_obj.force_fitting = DEFAULT_TRAINING_SETUP['force_fitting']
+            step_obj.persist_output = DEFAULT_TRAINING_SETUP['persist_output']
+            step_obj.cache_output = DEFAULT_TRAINING_SETUP['cache_output']
+            step_obj.load_persisted_output = DEFAULT_TRAINING_SETUP['load_persisted_output']
+            step_obj.persist_upstream_pipeline_structure = DEFAULT_TRAINING_SETUP['persist_upstream_pipeline_structure']
+        logger.info('Step {}, reset all upstream Steps to default training parameters, '
+                    'including this Step'.format(self.name))
+
+    def set_parameters_upstream_steps(self, parameters):
+        """Set parameters to all upstream Steps including this Step.
+        Parameters is dict() where key is Step attribute, and value is new value to set.
+        """
+        assert isinstance(parameters, dict), 'parameters must be dict, got {} instead'.format(type(parameters))
+        for name, step_obj in self.all_upstream_steps.items():
```

Comment:
@kamil-kaczmarek again name is not used anywhere in this loop. Probably one should use .values() .

Reply:
corrected